lib/redis_client/ruby_connection/buffered_io.rb

```diff
@@ -174,7 +171,7 @@ def fill_buffer(strict, size = @chunk_size)
             if empty_buffer
               @offset = start
               empty_buffer = false
-              @buffer.force_encoding(Encoding::UTF_8) if RESET_BUFFER_ENCODING
+              @buffer.force_encoding(Encoding::UTF_8) unless @buffer.encoding == Encoding::UTF_8
```

Comment:
Yeah, I wanted to avoid this check. I know it's likely very minimal overhead, but when trying to be get the Ruby implementation competitive with the C one, all these small things kinda count.

But OK, let's merge that for now, and then I'll see if I can feature test this in some way.